graph/db/sql_store.go

```diff
@@ -80,3 +131,749 @@ func NewSQLStore(db BatchedSQLQueries, kvStore *KVStore,
 
 	return s, nil
 }
+
+// TxOptions defines the set of db txn options the SQLQueries
+// understands.
+type TxOptions struct {
+	// readOnly governs if a read only transaction is needed or not.
+	readOnly bool
+}
+
+// ReadOnly returns true if the transaction should be read only.
+//
+// NOTE: This implements the TxOptions.
+func (a *TxOptions) ReadOnly() bool {
+	return a.readOnly
+}
+
+// NewReadTx creates a new read transaction option set.
+func NewReadTx() TxOptions {
```

Comment:
We could return a pointer here, then we could use `NewReadTx()` directly in the call to `ExecTx`.
And we could move that to the `sqldb` for re-use (we define too many different types of them in `tapd` for example with no real reason). And perhaps shorten to `ReadTx()` and also add a `WriteTx()`?